### PR TITLE
Issue/129 selsymb

### PR DESCRIPTION
**Pull request by vitalych**

### CODE REVIEW

File: common/s2ecmd/s2ecmd.cpp [hunk 1]
```diff
@@ -48,13 +48,15 @@
 typedef int (*cmd_handler_t)(int argc, const char **args);
 
 typedef struct _cmd_t {
-    char *name;
+    const char *name;
```
Comment (adrianherrera) [inline common/s2ecmd/s2ecmd.cpp:51]: Does this have to be a cpp file? You should still be able to compile c and cop files together. I can’t see anything here that is really c++ specific

File: common/s2ecmd/symfile.cpp [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+
+#ifdef _WIN32
+    flags |= O_BINARY;
+#endif
+
+    symbolic_locs_t sym_ranges;
+    bitmap_t sym_bitmap;
+
+    // TODO: implement proper command line args parsing
+    const char *sym_ranges_env = getenv("SYMFILE_RANGES");
```
Comment (adrianherrera) [inline common/s2ecmd/symfile.cpp]: The environment variable here (SYMFILE_RANGES) does not match with what’s in the comments (SYM_FILE_RANGES).

Additionally, I think you should prepend whichever one you choose with S2E_

Comment (vitalych) [reply common/s2ecmd/symfile.cpp]: nice catch!

File: common/s2ecmd/s2ecmd.cpp (hunk 1 above)
Comment (vitalych) [reply common/s2ecmd/s2ecmd.cpp:51]: I think it's actually better to compile in c++ mode even c files. That's how it found the const bug above.

File: common/s2ecmd/symfile.cpp [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+#endif
+
+    symbolic_locs_t sym_ranges;
+    bitmap_t sym_bitmap;
+
+    // TODO: implement proper command line args parsing
+    const char *sym_ranges_env = getenv("S2E_SYMFILE_RANGES");
+    if (sym_ranges_env) {
+        if (!parse_symbolic_ranges(sym_ranges_env, sym_ranges)) {
+            s2e_kill_state_printf(0, "Invalid SYMFILE_RANGES variable: %s", sym_ranges_env);
```
Comment (adrianherrera) [inline common/s2ecmd/symfile.cpp]: Can you change the message to S2E_

File: common/s2ecmd/s2ecmd.cpp [hunk 4]
```diff
@@ -426,14 +318,19 @@ static int find_command(const char *cmd) {
 }
 
 int main(int argc, const char **argv) {
+    const char *cmd = NULL;
```
Comment (adrianherrera) [inline common/s2ecmd/s2ecmd.cpp]: Since this is C++ code now, you should probably `s/NULL/nullptr/g`

Comment (vitalych) [reply common/s2ecmd/s2ecmd.cpp]: good idea!

File: common/s2ecmd/symfile.cpp (hunk 3 above)
Comment (vitalych) [reply common/s2ecmd/symfile.cpp]: I missed that sorry

**Review by adrianherrera — APPROVED** [review]
Lgtm

**Pull request merged**